hyperspy/io_plugins/hspy.py

```diff
@@ -544,6 +544,9 @@ def overwrite_dataset(group, data, key, signal_axes=None, chunks=None, **kwds):
             # Optimise the chunking to contain at least one signal per chunk
             chunks = get_signal_chunks(data.shape, data.dtype, signal_axes)
 
+    if np.issubdtype(data.dtype, np.dtype('U')):
+        # Saving numpy unicode type is not supported in h5py
+        data = data.astype(np.dtype('S'))
```

Comment:
I notice that `'S'` zero-terminated bytes  are ["not recommended"](https://numpy.org/doc/stable/reference/arrays.dtypes.html) according to the numpy docs. Did you check to see if there are any alternatives here?

Reply:
Yes, I have read this too and I think this is fine because this is used only to write hdf5 dataset.